File: Sort_Array_By_Parity_II/solution.best.py
class Solution:
  def sortArrayByParityII(self, A: List[int]) -> List[int]:
    ans = [ 0 ] * len(A)
    evenIdx = 0
    oddIdx = 1
    for v in A:
      if v % 2 == 0:
        ans[evenIdx] = v
        evenIdx += 2
      else:
        ans[oddIdx] = v
        oddIdx += 2
    return ans
    

File: Sort_Array_By_Parity_II/solution.best.java
class Solution {
  public int[] sortArrayByParityII(int[] A) {
    int ans[] = new int[A.length];
    int evenIdx = 0;
    int oddIdx = 1;
    for (int v : A) {
      if (v % 2 == 0) {
        ans[evenIdx] = v;
        evenIdx += 2;
      } else {
        ans[oddIdx] = v;
        oddIdx += 2;
      }
    }
    return ans;
  }
}
